src/controllers/restaurants.contorller.ts

```diff
@@ -0,0 +1,50 @@
+import { Request, Response } from 'express';
+import * as restaurantHandler from '../handlers/restaurants.handler';
+
+export const addNewRestaurant = async (req: Request, res: Response) => {
+    try {
+        const newRestaurant = await restaurantHandler.addNewRestaurant(req.body);
+        res.status(201).json(newRestaurant);
+    } catch (error: any) {
+        res.status(500).json({ message: error.message });
+    }
+};
+
+export const getAllRestaurants = async (req: Request, res: Response) => {
+    try {
```

Comment:
Same here, you are bringing the restaurants that are deleted, please filter them here and in getRestaurantById (in that function you will need to return 400 if the restaurant was deleted)